cps/kni.c

```diff
@@ -485,3 +486,314 @@ rm_kni(void)
 		RTE_LOG(ERR, GATEKEEPER, "Error removing %s: %s\n",
 			name, strerror(errno));
 }
+
+/*
+ * Processing ARP and ND packets from the KNI. If responding to
+ * an ARP/ND packet fails, we remove the request from the linked list,
+ * forcing the KNI to issue another resolution request.
+ */
+
+static void
+cps_arp_cb(const struct lls_map *map, void *arg,
+	__attribute__((unused)) enum lls_reply_ty ty, int *pcall_again)
+{
+	struct gatekeeper_if *iface = arg;
+	struct rte_mbuf *created_pkt;
+	struct ether_hdr *eth_hdr;
+	struct arp_hdr *arp_hdr;
+	size_t pkt_size;
+	struct cps_config *cps_conf = get_cps_conf();
+	struct rte_kni *kni;
+	struct rte_mempool *mp;
+	struct cps_request *req;
+	int ret;
+
+	if (pcall_again != NULL)
+		*pcall_again = false;
```

Comment:
Adding `RTE_VERIFY(!map->stale)` after the `pcall_again` test will guarantee that we don't send improper replies.